hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -138,6 +138,20 @@ def trivial(self):
         return self.forced or self.all_zero
 
 
+@attr.s(frozen=True)
+class OverrunPlaceholder(object):
+    """A dummy object that looks just enough like a ConjectureData
+    instance to be used as placeholder return values for circumstances
+    where we can tell that a test function call would overrun without
+    bothering to actually run it."""
+
+    status = Status.OVERRUN
+    blocks = ()
+    examples = ()
```

Comment:
I think it's possible to get rid of these fake block/example lists, as long as we add an explicit overrun check to `zero_examples` just above the existing index check.

That seems safer to me than using a fake example list to “fool” the index check, and makes it easier to detect future passes that accidentally try to read absent block/example information.

Reply:
Yeah I agree. Actually I think we can remove the buffer too and make this uniformly a single placeholder object.